### ISSUE TITLE
Padded action dimensions can affect valid predictions in GR00T N1.7

### ISSUE THREAD

**Issue by Henry-23** [NVIDIA/Isaac-GR00T#742]
### 🐛 Describe the bug

GR00T N1.7 uses fixed-size action tensors for cross-embodiment training, but padded action values are masked only when computing the final loss.

During training, Gaussian noise is sampled for the full action tensor:

```python
noise = torch.randn(actions.shape)
noisy_trajectory = (1 - t) * noise + t * actions
action_features = self.action_encoder(noisy_trajectory, ...)
```

The dataset padding is zero, but its noisy value is `(1 - t) * noise`. These padded values enter the action encoder before the loss mask is applied.

Fully padded action timesteps also enter DiT self-attention as normal tokens, so they can change valid action predictions.

This minimal example reproduces the DiT token leakage without data or a checkpoint:

```python
import torch

from gr00t.model.modules.dit import DiT


torch.manual_seed(7)

model = DiT(
    num_attention_heads=2,
    attention_head_dim=4,
    output_dim=8,
    num_layers=4,
    dropout=0.0,
    final_dropout=False,
    interleave_self_attention=True,
    cross_attention_dim=8,
).eval()

valid = torch.randn(1, 4, 8)
padded = torch.cat([valid, torch.randn(1, 2, 8)], dim=1)
encoder = torch.randn(1, 5, 8)
timestep = torch.tensor([3])

compact_output = model(valid, encoder, timestep)
padded_output = model(padded, encoder, timestep)[:, :4]

max_diff = (compact_output - padded_output).abs().max().item()
print(f"max_abs_diff={max_diff:.9f}")

assert max_diff < 1e-6, (
    f"Padded tokens changed valid outputs: max_abs_diff={max_diff}"
)
```

Observed result: `max_abs_diff` is non-zero and the assertion fails. The valid output changes even though only padded tokens were added.

Expected behavior: padded action dimensions and timesteps should not affect valid predictions. The action mask should be applied before action encoding, padded horizon tokens should be excluded from DiT self-attention, and padded values should remain zero during inference and Euler integration.

### Proposed fix

A proposed fix is available in #741:

- Apply the action mask before action encoding.
- Exclude padded horizon tokens from DiT self-attention.
- Keep padded values zero during Euler integration.
- Extend the inference mask to cover both horizon and action dimensions.
- Update the ONNX and TensorRT paths.

Related PR: https://github.com/NVIDIA/Isaac-GR00T/pull/741

### Versions

Isaac-GR00T commit:
b9955401d50c92a29258732e3ad6ccd579f1bdc0

Python:
Python 3.10.12

Relevant pip freeze output:
accelerate==1.13.0
diffusers==0.35.1
einops==0.8.1
flash-attn @ https://github.com/Dao-AILab/flash-attention/releases/download/v2.7.4.post1/flash_attn-2.7.4.post1+cu12torch2.7cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
huggingface-hub==0.36.2
numpy==1.26.4
peft==0.17.1
safetensors==0.7.0
torch==2.7.1+cu128
torchcodec==0.4.0
torchvision==0.22.1+cu128
transformers==4.57.3
triton==3.3.1

**Comment by jiajingc-nv** [comment]
**Open question on this issue: what should `strict_action_padding_mask` default to?**

#741 does fix the leak, but as of 1adbaa5 the fix sits behind that flag, and the flag defaults to `False` in the model config, the processor config, and the finetune CLI. So if #741 merges as-is, the repro at the top of this issue still reproduces on `main` by default, in training as well as at inference. That is worth deciding deliberately rather than by omission, because #741 carries `Fixes #742` and would auto-close this issue on merge.

**Why off is right for part of the problem.** Every released checkpoint was trained with the padding attending and has been served that way, so masking it at serve time is a distribution shift against those weights. For `libero_sim` on `nvidia/GR00T-N1.7-3B` (`action_horizon: 40`, `max_action_dim: 132`, 16 valid horizon steps over 7 action keys), that means 24 of 40 horizon tokens dropping out of DiT self-attention and 125 of 132 action dims zeroed. Not a marginal change to force onto existing checkpoints.

**Why it may be broader than it needs to be.** A newly trained model has no released-weight compatibility at stake, and the flag is persisted into both `config.json` and `processor_config.json`, so a model trained with it on is also served with it on. One boolean currently covers both cases, so new training inherits a restriction that only the released-checkpoint case needs.

The honest counter-argument: most people finetune from `nvidia/GR00T-N1.7-3B` rather than training from scratch, and finetuning legacy weights under strict masking is itself a mild mismatch. Whether that helps or hurts is an evaluation question, not one that can be settled by argument, and I cannot run it locally.

Either resolution looks defensible:

- evaluate a released checkpoint with the flag on and set the default from the result; or
- keep the default off, change #741 to `Refs #742`, and leave this issue open until a checkpoint trained under the flag exists.

Related work: Henry-23/Isaac-GR00T#2 closes the remaining seams around the flag on top of #741. It adds the serving-side agreement check between the model and processor configs, pins `action_mask` survival through the collator, exposes the flag on the finetune CLI (it was unreachable from `launch_finetune.py`, which copies model settings field by field), and documents it. It deliberately does not change the default.
